### PR TITLE
Admin logs in #37

### PR DESCRIPTION
**Pull request by brennanholtzclaw**
User story:
As an Admin
      When I log in
      Then I am redirected to "/admin/dashboard"

As an Admin
      When I visit "/admin/dashboard"
      I will see a heading on the page that says "Admin Dashboard"
      As a registered user
      When I visit "/admin/dashboard"
      I get a 404
      As an unregistered user
      When I visit "/admin/dashboard"
      I get a 404

### CODE REVIEW

**Comment by brennanholtzclaw** [comment]
We should try to config Hound to get rid of some errors. Rachel said it was OK to ignore the two that are still here.

File: app/controllers/admin/base_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Admin::BaseController < ApplicationController
+  before_action :require_admin
+
+  def require_admin
+    render file: "public/404" unless current_admin?
```
Comment (erinnachen) [inline app/controllers/admin/base_controller.rb]: we could think about less harsh redirects... although I do like that sweet 404 page.

File: config/routes.rb [hunk 2]
```diff
@@ -9,6 +9,10 @@
   resources :orders, only: [:index, :create, :show]
   get "/dashboard", to: "users#show", as: "dashboard"
 
+  namespace :admin do
+    resources :dashboard, only: [:index], as: "user"
```
Comment (erinnachen) [inline config/routes.rb]: interesting... it's not obvious to me why it needs `as: "user"` here but I think it's fine for now, the helper path seems cleaner.

File: spec/features/admin_sees_dashboard_users_do_not_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    expect(page).to have_content("Out of Luck!")
+  end
+  scenario "visitor gets 404 on admin dashboard" do
+    visit "/admin/dashboard"
+
+    expect(page).to have_content("Out of Luck!")
+  end
+end
+# As an Admin
```
Comment (erinnachen) [inline spec/features/admin_sees_dashboard_users_do_not_spec.rb]: Can we take these comments out prior to merging to master?

Comment (brennanholtzclaw) [inline spec/features/admin_sees_dashboard_users_do_not_spec.rb]: oops, thanks!

File: config/routes.rb (hunk 2 above)
Comment (brennanholtzclaw) [inline config/routes.rb]: This was something I changed while tests were breaking - never thought to change it back. After fiddling with it, it looks like you don't need that bit. Changed it and changed the path helper in the one test where it's called.

**Comment by erinnachen** [comment]
Good stuff.

**Comment by ksk5280** [comment]
Sounds good Brennan, please make an issue card when you come across something we need to tackle so that can remember to get to it! I'll make a card for this hound issue

**Pull request merged**